irohad/ametsuchi/key_value_storage.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_KV_STORAGE_HPP
+#define IROHA_KV_STORAGE_HPP
+
+#include <boost/optional.hpp>
+#include <string>
+#include <vector>
+
+namespace iroha {
+
+  namespace ametsuchi {
+
+    /**
+     * Solid storage interface
+     */
+    class KeyValueStorage {
+     public:
+      /**
+       * Type of storage key
+       */
+      using Identifier = uint32_t;
+
+      /**
+       * Add entity with binary data
+       * @param id - reference key
+       * @param blob - data associated with key
+       */
+      virtual bool add(Identifier id, const std::vector<uint8_t> &blob) = 0;
```

Comment:
In shared_model/cryptography/blob.hpp we have Bytes type, which is std::vector<uint8_t>. Maybe use it here? I am not sure about adding a dependency to shared model, but we can make it a separate declaration.